src/standalone/Soil/energy_hydrology.jl

```diff
@@ -1167,119 +1097,35 @@ function ClimaLand.surface_specific_humidity(
-the fraction of the humidity in the air due to ice, as
 
-f = S_i/(S_i+S_l)
-
-This same fraction is used to estimate the specific humidity, i.e.
-q = q_over_ice * f + q_over_water * (1-f).
-"""
-function ice_fraction(θ_l::FT, θ_i::FT, ν::FT, θ_r::FT)::FT where {FT}
-    S_l = effective_saturation(ν, θ_l, θ_r)
-    S_i = effective_saturation(ν, θ_i, θ_r)
-    f = S_i / (S_i + S_l)
-    return f
-end
-
-"""
-    soil_tortuosity(θ_l::FT, θ_i::FT, ν::FT) where {FT}
-
-Computes the tortuosity of water vapor in a porous medium,
-as a function of porosity `ν` and the volumetric liquid water
-and ice contents, `θ_l` and `θ_i`.
-
-See Equation (1) of : Shokri, N., P. Lehmann, and
-D. Or (2008), Effects of hydrophobic layers on evaporation from
-porous media, Geophys. Res. Lett., 35, L19407, doi:10.1029/
-2008GL035230.
-"""
-function soil_tortuosity(θ_l::FT, θ_i::FT, ν::FT) where {FT}
-    safe_θ_a = max(ν - θ_l - θ_i, eps(FT))
-    return safe_θ_a^(FT(2.5)) / ν
-end
-
-"""
-    soil_resistance(θ_l::FT,
-                    θ_i::FT,
-                    hydrology_cm::C,
-                    ν::FT,
-                    θ_r::FT,
-                    d_ds::FT,
-                    earth_param_set::EP,
-                   ) where {FT, EP, C}
-
-Computes the resistance of the top of the soil column to
-water vapor diffusion, as a function of the surface
-volumetric liquid water fraction `θ_l`, the augmented
-liquid water fraction `ϑ_l`,  the volumetric ice water
-fraction `θ_i`, and other soil parameters.
-"""
-function soil_resistance(
-    θ_l::FT,
-    θ_i::FT,
-    hydrology_cm::C,
-    ν::FT,
-    θ_r::FT,
-    d_ds::FT,
-    earth_param_set::EP,
-) where {FT, EP, C}
-    (; S_c) = hydrology_cm
-    _D_vapor = FT(LP.D_vapor(earth_param_set))
-    S_w = effective_saturation(ν, θ_l + θ_i, θ_r)
-    τ_a = soil_tortuosity(θ_l, θ_i, ν)
-    dsl::FT = dry_soil_layer_thickness(S_w, S_c, d_ds)
-    r_soil = dsl / (_D_vapor * τ_a) # [s\m]
-    return r_soil
-end
-
-"""
-    dry_soil_layer_thickness(S_w::FT, S_c::FT, d_ds::FT)::FT where {FT}
-
-Returns the maximum dry soil layer thickness that can develop under vapor flux;
-this is used when computing the soil resistance to vapor flux according to
-Swenson et al (2012)/Sakaguchi and Zeng (2009).
-"""
-function dry_soil_layer_thickness(S_w::FT, S_c::FT, d_ds::FT)::FT where {FT}
-    return S_w < S_c ? d_ds * (S_c - S_w) / S_c : FT(0)
+    ice_frac = p.soil.ice_frac
+    @. ice_frac = ice_fraction(θ_l_sfc, θ_i_sfc, ν_sfc, θ_r_sfc, _ρ_l, _ρ_i)
+    @. p.soil.q_sfc =
+        (1 - ice_frac) *
+        Thermodynamics.q_vap_saturation_generic(
+            thermo_params,
+            T_sfc,
+            ρ_sfc,
+            Thermodynamics.Liquid(),
+        ) *
+        exp(g * ψ_sfc * M_w / (R * T_sfc)) +
+        ice_frac * Thermodynamics.q_vap_saturation_generic(
+            thermo_params,
+            T_sfc,
+            ρ_sfc,
+            Thermodynamics.Ice(),
```

Comment:
Minor thing, but you could collect `Thermodynamics.q_vap_saturation_generic` in this expression to compute it only once